src/serialized-utilities.cpp

```diff
@@ -0,0 +1,160 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+#include "serialized-utilities.h"
+#include <device.h>
+
+using json = nlohmann::json;
+
+namespace librealsense {
+namespace serialized_utilities {
+
+static const char* SCHEMA_VERSION = "1.0";
+
+json_preset_reader::json_preset_reader( const std::string & json_content ) : _parameters(nullptr)
+{
+    _root = json::parse( json_content );
+    if (validate_schema())
+    {
+        _schema_version = get_value(_root, "schema version").get<std::string>();
+        _device_info = read_device_info();
+        _parameters = &_root["parameters"];
+    }
+    else
+    {
+        _parameters = &_root;
+    }
+}
+
+device_info json_preset_reader::read_device_info() const
```

Comment:
Any time we want to `read_device_info()` and not set `_device_info`?

Reply:
Currently I use it to read modify write on the D400 side.
I can just set all fields but I like it better this way, It may be handy in the future to read it